### PR TITLE
LT-949-Toggle-Case-Flags

### PR DESCRIPTION
**Pull request by andytaylorkainos**

### CODE REVIEW

File: cases/views.py [hunk 1]
```diff
@@ -229,6 +227,35 @@ def post(self, request, **kwargs):
...
+    def get(self, request, **kwargs):
+        case_id = str(kwargs['pk'])
+        case_data, status_code = get_case(request, case_id)
+        case_level_team_flags_data, status_code = get_flags_case_level_for_team(request)
+        case_flags = case_data.get('case').get('flags')
+        case_level_team_flags = case_level_team_flags_data.get('flags')
+
+        for flag in case_level_team_flags:
+            for case_flag in case_flags:
+                flag['selected'] = flag['id'] == case_flag['id']
```
Comment (cstobart) [inline cases/views.py]: if flag['id'] == case_flag['id']

File: templates/cases/case/application-case.html [hunk 2]
```diff
@@ -30,6 +30,21 @@ <h1 class="govuk-heading-l">{{ title }}</h1>
 	</div>
 </div>
 
+<!-- # TODO: Create modal for clients with JS enabled -->
+<!-- <a href="{% url 'cases:assign_flags' data.case.id %}" class="govuk-link govuk-link--no-visited-state" onclick="return showFlagsModal();">Edit case flags</a> -->
+<a href="{% url 'cases:assign_flags' data.case.id %}" class="govuk-link govuk-link--no-visited-state">Edit case flags</a>
```
Comment (cstobart) [inline templates/cases/case/application-case.html]: strings

Comment (andytaylorkainos) [reply templates/cases/case/application-case.html]: Done

File: ui_automation_tests/step_defs/test_assign_case_flags_to_case.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+scenarios('../features/assign_case_flags_to_case.feature', strict_gherkin=False)
+
+log = logging.getLogger()
+console = logging.StreamHandler()
+log.addHandler(console)
+
+
+# TODO: Replace with API fixture to create flags to remove test dependency
+@given('Case flags have been created')
```
Comment (phlptt) [inline ui_automation_tests/step_defs/test_assign_case_flags_to_case.py]: doesnt look like this is being used

File: ui_automation_tests/step_defs/test_assign_case_flags_to_case.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+
+@then("Number of assigned flags is original value")
+def assert_number_of_flags(driver):
+    number_of_assigned_flags = len(driver.find_elements_by_class_name("lite-flag"))
+    assert number_of_assigned_flags == context.number_of_assigned_flags
+
+
+@then("Number of assigned flags has increased")
+def assert_number_of_flags(driver):
```
Comment (phlptt) [inline ui_automation_tests/step_defs/test_assign_case_flags_to_case.py]: this function name needs to be different to the one previous

File: ui_automation_tests/step_defs/test_assign_case_flags_to_case.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    number_of_assigned_flags = len(driver.find_elements_by_class_name("lite-flag"))
```
Comment (phlptt) [inline ui_automation_tests/step_defs/test_assign_case_flags_to_case.py]: can we store this lite-flag class name in a page object as it is reused a few times

File: ui_automation_tests/step_defs/test_assign_case_flags_to_case.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+from pages.shared import Shared
+from pytest_bdd import given, when, then, scenarios
+from pages.flags_pages import FlagsPages
+from pages.assign_flags_to_case import CaseFlagsPages
+from pages.application_page import ApplicationPage
+from pages.header_page import HeaderPage
+
+scenarios('../features/assign_case_flags_to_case.feature', strict_gherkin=False)
+
+log = logging.getLogger()
```
Comment (phlptt) [inline ui_automation_tests/step_defs/test_assign_case_flags_to_case.py]: doesnt look like this is used

File: ui_automation_tests/step_defs/test_assign_case_flags_to_case.py [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    case_flags_pages = CaseFlagsPages(driver)
+    case_flags_pages.assign_flags(context)
+    shared = Shared(driver)
+    shared.click_submit()
+
+
+@then("Number of assigned flags is original value")
+def assert_number_of_flags(driver):
+    number_of_assigned_flags = len(driver.find_elements_by_class_name("lite-flag"))
+    assert number_of_assigned_flags == context.number_of_assigned_flags
```
Comment (phlptt) [inline ui_automation_tests/step_defs/test_assign_case_flags_to_case.py]: optional - add a message at the end for if the assert fails e.g.
assert number_of_assigned_flags == context.number_of_assigned_flags, "number of assigned flags was not expected"

File: ui_automation_tests/step_defs/test_assign_case_flags_to_case.py [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+
+@then("Number of assigned flags has increased")
+def assert_number_of_flags(driver):
+    number_of_assigned_flags = len(driver.find_elements_by_class_name("lite-flag"))
+    assert number_of_assigned_flags == context.number_of_assigned_flags + 1
```
Comment (phlptt) [inline ui_automation_tests/step_defs/test_assign_case_flags_to_case.py]: optional - add a message at the end for if the assert fails e.g.
assert number_of_assigned_flags == context.number_of_assigned_flags+1, "number of assigned flags has not changed after adding a new flag"

**Pull request merged**